pkg/controller/v1beta1/inferenceservice/utils/utils.go

```diff
@@ -163,3 +168,30 @@ func GetServingRuntime(cl client.Client, name string, namespace string) (*v1alph
 	}
 	return nil, goerrors.New("No ServingRuntimes or ClusterServingRuntimes with the name: " + name)
 }
+
+// Replace placeholders in runtime container by values from inferenceservice metadata
+func ReplacePlaceholders(container *v1.Container, meta metav1.ObjectMeta) error {
+	data, _ := json.Marshal(container)
+	tmpl, err := template.New("container-tmpl").Parse(string(data))
+	if err != nil {
+		return err
+	}
+	buf := &bytes.Buffer{}
+	err = tmpl.Execute(buf, meta)
+	if err != nil {
+		return err
+	}
+	return json.Unmarshal(buf.Bytes(), container)
+}
+
+// Update image tag if GPU is enabled or runtime version is provided
+func UpdateImageTag(container *v1.Container, runtimeVersion string) {
+	image := container.Image
+	if utils.IsGPUEnabled(container.Resources) &&
```

Comment:
Hmm, I'm wondering if we should have this gpu block. It might have the unintended side effect of appending `-gpu` to image tags that might not need it since, in this case, any inferenceservice with a `resources.limits` object containing `nvidia.com/gpu` will automatically have the `-gpu` appended.

I'm thinking maybe just keep it simple here, and just rely on the user passing in the correct runtimeVersion or using a runtime that already specifies a gpu-enabled image. WDYT?

Reply:
Agreed, for example triton does not differentiate cpu or gpu image while tfserving, torchserve does. So I think the logic should be following:
1) if `runtimeVersion` is specified, user should pass in the right gpu image version like `tfserving:1.14-gpu`
2) if `runtimeVersion` is not specified, we still get the default image version from the existing inference service configmap, so we do not need to add the kustomization layer to patch the image version for serving runtimes.